azure-pipelines.yml

```diff
@@ -144,6 +156,7 @@ jobs:
           mkdir -p .azure-cache
           tar czf .azure-cache/stack-root.tar.gz -C $HOME .stack
         displayName: "Pack cache"
+        condition: not(eq('true', variables['cacheFailed']))
```

Comment:
The uncached builds are quite slow here. Can we easily mark this as non-required? I couldn’t find anything in the Azure docs but maybe I’m just missing something.
If not, we probably need to disable this on Monday if nothing has changed until then.

Reply:
How slow? macOS builds typically take about 25 minutes, so as long as it's faster than that it should be OK. An easy way to speed those up would be to run them on our own linux pool, which has bigger machines than the Azure-provided ones. Another option would be to use the same logic as in this PR to skip the build entirely if caching fails.

Reply:
I think it was around 40 minutes.

Reply:
Looks, like I was misremembering. It finished in ~20 minutes so it’s probably fine. However, the workaround doesn’t seem to work since the cache upload step is failing.

Reply:
Looks like this takes about 15 minutes, so well below the macOS one.